arm-mediaservices/2015-10-01/swagger/media.json

```diff
@@ -0,0 +1,715 @@
+            "minLength": 3,
+            "maxLength": 24
+          }
+        ]
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Media/mediaservices/{mediaServiceName}/syncStorageKeys": {
+      "post": {
+        "summary": "Synchronize the keys for a storage account to the Media Service.",
+        "operationId": "MediaServices_SyncStorageKeys",
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/MediaService"
+            }
+          },
+          "202": {
+            "description": "Accepted"
+          },
+          "default": {
+            "description": "Detailed error information.",
+            "schema": {
+              "$ref": "#/definitions/ApiError"
+            }
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "name": "resourceGroupName",
+            "description": "Name of the resource group within the Azure subscription.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "mediaServiceName",
+            "description": "Name of the Media Service.",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-z0-9]",
+            "minLength": 3,
+            "maxLength": 24
+          },
+          {
+            "name": "SyncStorageKeysInput",
+            "description": "Properties needed to sycnronize the keys for a storage account to the Media Service.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SyncStorageKeysInput"
+            }
+          }
+        ]
+      }
+    }
+  },
+  "definitions": {
+    "ApiEndpoint": {
+      "description": "The properties for a Media Services REST API endpoint.",
+      "properties": {
+        "endpoint": {
+          "description": "The Media Services REST endpoint.",
+          "type": "string"
+        },
+        "majorVersion": {
+          "description": "The version of Media Services REST API.",
+          "type": "string"
+        }
+      }
+    },
+    "ApiError": {
+    "description": "The error returned from a failed Media Services REST API call.",
+      "properties": {
+        "code": {
+          "description": "Error code.",
+          "type": "string"
+        },
+        "message": {
+          "description": "Error message",
+          "type": "string"
+        }
+      }
+    },
+    "CheckNameAvailabilityInput": {
+      "description": "The request body for CheckNameAvailability API.",
+      "properties": {
+        "name": {
+          "description": "Specifies the type of the resource.",

```

Comment:
"Specifies the name of the mediaservice to check whether it is available". It would be nice to add the same validation constraints (regex, min and max length) that are provided for the mediaServiceName path parameter. It is better to fail fast rather than saying that the desired name is available and then fail on validation when the user is creating a media service.